docs/source/examples/layerwise-folding.md

```diff
@@ -0,0 +1,305 @@
+    noise_model = None
+else:
+    # Default to a simulator.
+    backend = qiskit.Aer.get_backend("qasm_simulator")
+    noise_model = initialized_depolarizing_noise(noise_level=0.02)
+
+shots = 10_000
+```
+
+**Note:** When `USE_REAL_HARDWARE` is set to `False`, a classically simulated noisy backend is used instead of a real quantum computer.
+
+## Helper functions
+
+```{code-cell} ipython3
+# Display circuit in a more visually appealing way.
+draw = lambda circuit: SVGCircuit(circuit)
+```
+
+The following function will return a list of circuits where the ith element in
+the list is a circuit with layer "i" folded `num_folds` number of times. This
+will be useful when analyzing how much folding increases the noise on a given
+layer.
+
+```{code-cell} ipython3
+def apply_num_folds_to_all_layers(circuit: cirq.Circuit, num_folds: int = 1) -> List[cirq.Circuit]:
+    """List of circuits where ith circuit is folded `num_folds` times."""
+    return [
+        layer_folding(circuit, [0] * i + [num_folds] + [0] * (len(circuit) - i))
+        for i in range(len(circuit))
+    ]
+```
+
+For instance, consider the following circuit.
+
+```{code-cell} ipython3
+# Define a basic circuit for
+q0, q1 = cirq.LineQubit.range(2)
+circuit = cirq.Circuit(
+    [cirq.ops.H(q0)],
+    [cirq.ops.CNOT(q0, q1)],
+    [cirq.measure(cirq.LineQubit(0))],
+)
+draw(circuit)
+```
+
+Let us invoke the `apply_num_folds_to_all_layers` function as follows.
+
+```{code-cell} ipython3
+folded_circuits = apply_num_folds_to_all_layers(circuit, num_folds=2)
+```
+
+Note that the first element of the list is the circuit with the first layer of
+the circuit folded twice.
+
+```{code-cell} ipython3
+draw(folded_circuits[0])
+```
+
+Similarly, the second element of the list is the circuit with the second layer
+folded.
+
+```{code-cell} ipython3
+draw(folded_circuits[1])
+```
+
+
+## Define circuit to analyze
+
+We will use the following circuit to analyze, but of course, you could use
+other circuits here as well.
+
+```{code-cell} ipython3
+circuit = cirq.Circuit([cirq.X(cirq.LineQubit(0))] * 10, cirq.measure(cirq.LineQubit(0)))
+draw(circuit)
+```
+
+## Total variational distance metric
+
+An $i$-inversion can be viewed as a local perturbation of the circuit. We want
+to define some measure by which we can determine how much such a perturbation
+affects the outcome.
+
+Define the quantity:
+
+$$
+p(k|C) = \langle \langle k | C | \rho_0 \rangle \rangle
+$$
+
+as the probability distribution oveer measurement outcomes at the output of a
+circuit $C$ where $k \in B^n$ with $B^n$ being the set of all $n$-length bit
+strings where $\langle \langle k |$ is the vectorized (non-ideal) POVM element
+that corresponds to measuring bit string $k$. 
+
+The *impact* of applying an inversion is given by
+
+$$
+d \left[p(\cdot|C), p(\cdot|C^{(i)})\right]
+$$
+
+where $d$ is some distance measure. In
+[arXiv:2204.06056](https://arxiv.org/abs/2204.06056), the authors used the
```

Comment:
I would cite the bib item of the same paper. We can still keep also the explicit arxiv link, if you think it is useful.

Reply:
Cited here as cited elsewhere in the ZNE section. Thanks!